Makefile

```diff
@@ -30,8 +30,12 @@ man: doc/downgrade.8
 test:
 	cram test
 
+.PHONY: config
+config:
```

Comment:
I think this should not be a `PHONY`, as it's building `downgrade` from `dowgrade.in`

```make
downgrade: downgrade.in
        sed ... $< > $@
```

And all the other tasks that you made depend on `downgrade.in`, they should depend on `downgrade` IMO, so that is is brought up to date and used in its interpolated form whenever needed.

Also, I usually use `m4` for this, but I honestly don't know why. It just felt more neck-beard-y.

Reply:
Yes I had tried this earlier, my arguments for making it a phony is the following edge-case:

Suppose one is doing local development and already created `downgrade` from a previous run. Then, when running `make install` with a non-default prefix (such as `/opt`), `downgrade` will not be made since `downgrade.in` was not modified. Now, `downgrade` will be installed with the "wrong" prefix. I just find it safer to always be able to remake `downgrade`, so any changes in the `make` prefix will always be reflected.

If we don't do this, we would always have to remember to delete a previously made version of `downgrade` if we want to experiment installing to another prefix.

Reply:
IMO the actual `downgrade` file is only important for installation. Everything else can work off `downgrade.in`.

Reply:
That's an interesting case. It just feels wrong to be using `downgrade.in` for things and having a phony target that always updates `downgrade`, whether it needs it or not. I'll give this some thought :thinking: 

I think local testing with alternative `PREFIX` is super unlikely, but there's also no real reason for worrying about making this sufficiently lazy.

Reply:
You mentioned something that I found compelling,

> IMO the [`PREFIX`] file is only important for installation

I think this is telling me to prefer the other option, where we create a file at installation time which contains the appropriate `TEXTDOMAINDIR` value, and then we source it from `downgrade`. Is there a reason you were against that?

I think I mentioned `/etc/profile.d/downgrade.sh` originally when I suggested that, but I'd also be fine with `$(PREFIX)/downgrade/textdomaindir.env` or something too.

Reply:
Ah yes you mentioned this. My issue with that was that it involves another file being installed. But then again this would fix the whole conundrum with `downgrade.in`.

Sounds like a better alternative. We can add the command to create this within the `install` phony, so it is all compact and we don't need another rule. We can update the `uninstalll` phony to remove it too. Just to be safe, we could add a condition in (the new) `downgrade` where `TEXTDOMAINDIR` defaults to some predefined value in case `/etc/profile.d/downgrade.sh` cannot be found/sourced, just to ensure the script does not fail.

Alright, I will sign off for the day. Need to powerpoint an e-pub-quiz for getting drunk with friends tomorrow!